modules/collector-source/pkg/collector/dcgmtargetprovider.go

```diff
@@ -0,0 +1,48 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/opencost/opencost/core/pkg/log"
+	"github.com/opencost/opencost/modules/collector-source/pkg/metrics/target"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type DCGMTargetProvider struct {
+	kubeClientSet kubernetes.Interface
+}
+
+func NewDCGMTargetProvider(kubeClientSet kubernetes.Interface) *DCGMTargetProvider {
+	return &DCGMTargetProvider{
+		kubeClientSet: kubeClientSet,
+	}
+}
+
+func (p *DCGMTargetProvider) GetTargets() []target.ScrapeTarget {
+	k8s := p.kubeClientSet
+
+	// Find service
+	svcs, err := k8s.CoreV1().Services("").List(context.Background(), metav1.ListOptions{
+		LabelSelector: "app.kubernetes.io/component=dcgm-exporter",
```

Comment:
Could we bring this LabelSelector into alignment with the following? https://github.com/kubecost/cost-analyzer-helm-chart/blob/5c51dc7b3cd91b124f904bbfc6eb362c1df3c08d/cost-analyzer/values.yaml#L880-L893

Depending on how you deploy DCGM Exporter, there are different label names and values that are used for the Service.

Reply:
Great callout @thomasvn

Reply:
I have left a comment with a TODO on this